UndertaleModLib/Util/TextureWorker.cs

```diff
@@ -1,217 +1,250 @@
-            SaveImageToFile(FullPath, blackImage);
-        }
+            // Get black/white colors to use for drawing
+            ReadOnlySpan<byte> black = MagickColor.FromRgba(0, 0, 0, 255).ToByteArray().AsSpan();
+            ReadOnlySpan<byte> white = MagickColor.FromRgba(255, 255, 255, 255).ToByteArray().AsSpan();
 
-        public static Bitmap GetCollisionMaskImage(UndertaleSprite sprite, UndertaleSprite.MaskEntry mask)
-        {
+            // Draw white pixels if a given bit is set; black pixels otherwise
             byte[] maskData = mask.Data;
-            Bitmap bitmap = new Bitmap((int)sprite.Width, (int)sprite.Height, PixelFormat.Format32bppArgb); // Ugh. I want to use 1bpp, but for some BS reason C# doesn't allow SetPixel in that mode.
-
             for (int y = 0; y < sprite.Height; y++)
             {
                 int rowStart = y * (int)((sprite.Width + 7) / 8);
                 for (int x = 0; x < sprite.Width; x++)
                 {
                     byte temp = maskData[rowStart + (x / 8)];
                     bool pixelBit = (temp & (0b1 << (7 - (x % 8)))) != 0b0;
-                    bitmap.SetPixel(x, y, pixelBit ? Color.White : Color.Black);
+                    pixels.SetPixel(x, y, pixelBit ? white : black);
                 }
             }
 
             return bitmap;
         }
 
-        public static void ExportCollisionMaskPNG(UndertaleSprite sprite, UndertaleSprite.MaskEntry mask, string fullPath)
+        /// <summary>
+        /// Exports a collision mask entry from the given sprite, as a PNG file, at the specified path, and with the given width/height.
+        /// </summary>
+        public static void ExportCollisionMaskPNG(UndertaleSprite sprite, UndertaleSprite.MaskEntry mask, string path, int maskWidth, int maskHeight)
         {
-            SaveImageToFile(fullPath, GetCollisionMaskImage(sprite, mask));
+            using var image = GetCollisionMaskImage(sprite, mask, maskWidth, maskHeight);
+            SaveImageToFile(image, path);
         }
 
-        public static byte[] GetImageBytes(Image image, bool disposeImage = true)
+        /// <summary>
+        /// Saves the provided image as a PNG file, at the specified path.
+        /// </summary>
+        public static void SaveImageToFile(IMagickImage<byte> image, string path)
         {
-            using (var ms = new MemoryStream())
-            {
-                image.Save(ms, image.RawFormat);
-                byte[] result = ms.ToArray();
-                if (disposeImage)
-                    image.Dispose();
-                return result;
-            }
+            using var stream = new FileStream(path, FileMode.Create);
+            image.Write(stream, MagickFormat.Png32);
         }
 
-        public static void SaveImageToFile(string FullPath, Image image, Boolean disposeImage = true)
+        /// <summary>
+        /// This should be called when the TextureWorker is no longer going to be used; this frees image data.
```

Comment:
Surely we can just use inheritdoc for this, no?
If not:

```suggestion
        /// This should be called when the TextureWorker is no longer going to be used, as it frees cached image data.
```